File: common/src/main/java/br/com/gustavoterras/foo/lib/common/DataBindingUtils.java
package br.com.gustavoterras.foo.lib.common;

import android.databinding.BindingAdapter;
import android.view.View;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;

public class DataBindingUtils {

    @BindingAdapter("fadeView")
    public static void fadeView(final View view, final Boolean show) {

        if (show && view.getVisibility() == View.VISIBLE) return;

        Animation anim = AnimationUtils.loadAnimation(view.getContext(), show ? android.R.anim.fade_in : android.R.anim.fade_out);

        anim.setAnimationListener(new Animation.AnimationListener() {
            @Override
            public void onAnimationStart(Animation animation) {
                if (show)
                    view.setVisibility(View.VISIBLE);
            }

            @Override
            public void onAnimationEnd(Animation animation) {
                if (!show)
                    view.setVisibility(View.INVISIBLE);
            }

            @Override
            public void onAnimationRepeat(Animation animation) {

            }
        });

        view.startAnimation(anim);
    }
}

File: common/src/main/java/br/com/gustavoterras/foo/lib/common/DataBindingUtils.kt
package br.com.gustavoterras.foo.lib.common

import android.databinding.BindingAdapter
import android.view.View
import android.view.animation.Animation
import android.view.animation.AnimationUtils

object DataBindingUtils {

    @BindingAdapter("fadeView")
    fun fadeView(view: View, show: Boolean?) {

        if (show!! && view.visibility == View.VISIBLE) return

        val anim = AnimationUtils.loadAnimation(view.context, if (show) android.R.anim.fade_in else android.R.anim.fade_out)

        anim.setAnimationListener(object : Animation.AnimationListener {
            override fun onAnimationStart(animation: Animation) {
                if (show)
                    view.visibility = View.VISIBLE
            }

            override fun onAnimationEnd(animation: Animation) {
                if (!show)
                    view.visibility = View.INVISIBLE
            }

            override fun onAnimationRepeat(animation: Animation) {

            }
        })

        view.startAnimation(anim)
    }
}
